client/src/geoUtils.js

```diff
@@ -0,0 +1,53 @@
+import { forward, toPoint } from "mgrs"
+
+export function parseCoordinate(latLng) {
+  const value = parseFloat(latLng)
+  if (!value && value !== 0) {
+    return null
+  }
+
+  /*
+   * We use 5 decimal point (~110cm) precision because MGRS has
+   * a minimum of 1 meter precision.
+   * Please see;
+   * https://stackoverflow.com/a/16743805/1209097
+   * https://en.wikipedia.org/wiki/Military_Grid_Reference_System
+   *
+   * Additionally, coordinates must be truncated instead of rounding when changing
+   * precision level in order to aviod inconsistencies during (MGRS <--> Lat/Lon) conversion.
+   */
+  const precision = 5
+  /*
+   * for the purpose of rounding below please see:
+   * https://stackoverflow.com/questions/1458633/how-to-deal-with-floating-point-number-precision-in-javascript
+   * https://floating-point-gui.de/
+   */
+  const safeRoundedValue = Math.round(value * 10 ** precision * 10) / 10
+  return Math.trunc(safeRoundedValue) / 10 ** precision
+}
```

Comment:
Unfortunately I can't see the history in this branch , but if I remember it correctly it was @VassilIordanov who made changes to the original implementation for this.
@VassilIordanov do you agree with this change?

Reply:
@VassilIordanov agreed with the change.